src/SlackRTMHandler.ts

```diff
@@ -275,17 +275,20 @@ export class SlackRTMHandler extends SlackEventHandler {
                     id ? this.main.ghostStore.get(id, event.team_domain, puppet.teamId) : null,
             ))).filter((g) => g !== null) as SlackGhost[];
 
-            const ghost = await this.main.ghostStore.getForSlackMessage(event, puppet.teamId);
+            const ghostIdx = ghosts.findIndex((g) => g.slackId !== puppet.slackId);
+            const ghost = ghosts[ghostIdx];
+            const otherGhosts = ghosts.slice(0, ghostIdx).concat(ghosts.slice(ghostIdx+1));
```

Comment:
> I wonder if we need to do this metadata fixing on something else than bridge startup: maybe m.room.member events? Or will this name mismatch never happen organically, and the startup code is just there to fix the rooms that are currently broken, because they were created before this change went in?

My understanding is that it's the latter.